src/routes/sessions/[id]/+page.svelte

```diff
@@ -202,6 +200,39 @@
 		session.messages = [...session.messages, message];
 	}
 
+	function resetPrompt() {
+		// Reset the prompt to the last sent message
+		prompt = promptCached;
+		promptCached = '';
+	}
+
+	function stopResponse(navigation?: BeforeNavigate) {
+		if (isWaitingForResponse) {
+			toast.warning(
+				"Can't stop completion until the first response from Ollama is received, please wait..."
+			);
+			navigation?.cancel();
+			return;
+		}
+
+		resetPrompt();
+		ollamaInstance?.abort();
+		completion = '';
+		session.messages = session.messages.slice(0, -1); // Remove the "incomplete" AI response
+	}
+
+	beforeNavigate((navigation) => {
+		if (!completion) return;
+		const userConfirmed = confirm(
+			'Are you sure you want to leave?\nThe completion in progress will stop'
+		);
+		if (!userConfirmed) {
+			navigation.cancel();
+			return;
+		}
+		stopResponse(navigation);
```

Comment:
Can we revert this syntax so it's less brittle?
```suggestion
		if (userConfirmed) {
			stopResponse(navigation);
			return;
		}
		navigation.cancel();
```